Externalized Templates Failing: Expected configuration ':swaggerTemplate' to contain exactly one file, however, it contains 2 files.

## Build Failing When Using Externalized Templates

I've externalized templates into another project, deployed to an Artifactory server. When I try to build I see the following error:
```
Could not determine the dependencies of task ':resolveSwaggerTemplate'.
> Expected configuration ':swaggerTemplate' to contain exactly one file, however, it contains 2 files.
```
Based on my testing deploying to Maven local, it appears to be an issue with the presence of a *.jar* and *pom.xml* file. When I manually deleted the `pom.xml` `resolveSwaggerTemplate` worked file. I think it has something to do with https://github.com/int128/gradle-swagger-generator-plugin/blob/master/src/main/groovy/org/hidetake/gradle/swagger/generator/ResolveSwaggerTemplate.groovy#L15 and it's use of `singleFile`

To Reproduce:
Publish a jar containing templates to Maven local, then use `mavenlocal()` to include that repository. Add the jar as a dependency using `swaggerTemplate` and the build will fail.

Expected behaviour:
The jar containing templates should be extracted to the *build* directory.

## Environment
Plugin version: 2.15.1
Swagger Codegen version: 2.4.0
Gradle version: 3.5.1
Java version: 1.8.0-121
OS: 10.12.6 (Sierra)


### Stacktrace
```
10:53:15 AM: Executing tasks 'clean build --stacktrace'...

Applying default Java plugins
Download http://my.artifactory/artifactory/local-gradle/foo/swagger-codegen-templates/1.1.0/swagger-codegen-templates-1.1.0.jar

FAILURE: Build failed with an exception.

* What went wrong:
Could not determine the dependencies of task ':resolveSwaggerTemplate'.
> Expected configuration ':swaggerTemplate' to contain exactly one file, however, it contains 2 files.

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.GradleException: Could not determine the dependencies of task ':resolveSwaggerTemplate'.
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:67)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.getDependencies(CachingTaskDependencyResolveContext.java:55)
	at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.addToTaskGraph(DefaultTaskExecutionPlan.java:139)
	at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.addTasks(DefaultTaskGraphExecuter.java:101)
	at org.gradle.execution.TaskNameResolvingBuildConfigurationAction.configure(TaskNameResolvingBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$000(DefaultBuildConfigurationActionExecuter.java:25)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.proceed(DefaultBuildConfigurationActionExecuter.java:54)
	at org.gradle.execution.DefaultTasksBuildExecutionAction.configure(DefaultTasksBuildExecutionAction.java:44)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.access$000(DefaultBuildConfigurationActionExecuter.java:25)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter$1.proceed(DefaultBuildConfigurationActionExecuter.java:54)
	at org.gradle.execution.ExcludedTaskFilteringBuildConfigurationAction.configure(ExcludedTaskFilteringBuildConfigurationAction.java:47)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.configure(DefaultBuildConfigurationActionExecuter.java:48)
	at org.gradle.execution.DefaultBuildConfigurationActionExecuter.select(DefaultBuildConfigurationActionExecuter.java:36)
	at org.gradle.initialization.DefaultGradleLauncher$CalculateTaskGraphAction.execute(DefaultGradleLauncher.java:220)
	at org.gradle.initialization.DefaultGradleLauncher$CalculateTaskGraphAction.execute(DefaultGradleLauncher.java:217)
	at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
	at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:158)
	at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:112)
	at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:95)
	at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:66)
	at org.gradle.tooling.internal.provider.runner.BuildModelActionRunner.run(BuildModelActionRunner.java:50)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.tooling.internal.provider.runner.RunAsBuildOperationBuildActionRunner$1.execute(RunAsBuildOperationBuildActionRunner.java:43)
	at org.gradle.tooling.internal.provider.runner.RunAsBuildOperationBuildActionRunner$1.execute(RunAsBuildOperationBuildActionRunner.java:40)
	at org.gradle.internal.Transformers$4.transform(Transformers.java:169)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:106)
	at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:56)
	at org.gradle.tooling.internal.provider.runner.RunAsBuildOperationBuildActionRunner.run(RunAsBuildOperationBuildActionRunner.java:40)
	at org.gradle.tooling.internal.provider.runner.SubscribableBuildActionRunner.run(SubscribableBuildActionRunner.java:88)
	at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
	at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:75)
	at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:49)
	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:49)
	at org.gradle.tooling.internal.provider.ServicesSetupBuildActionExecuter.execute(ServicesSetupBuildActionExecuter.java:31)
	at org.gradle.launcher.daemon.server.exec.ExecuteBuild.doBuild(ExecuteBuild.java:67)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.WatchForDisconnection.execute(WatchForDisconnection.java:37)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ResetDeprecationLogger.execute(ResetDeprecationLogger.java:26)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.RequestStopIfSingleUsedDaemon.execute(RequestStopIfSingleUsedDaemon.java:34)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:74)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput$2.call(ForwardClientInput.java:72)
	at org.gradle.util.Swapper.swap(Swapper.java:38)
	at org.gradle.launcher.daemon.server.exec.ForwardClientInput.execute(ForwardClientInput.java:72)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogAndCheckHealth.execute(LogAndCheckHealth.java:55)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.LogToClient.doBuild(LogToClient.java:60)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.EstablishBuildEnvironment.doBuild(EstablishBuildEnvironment.java:72)
	at org.gradle.launcher.daemon.server.exec.BuildCommandOnly.execute(BuildCommandOnly.java:36)
	at org.gradle.launcher.daemon.server.api.DaemonCommandExecution.proceed(DaemonCommandExecution.java:120)
	at org.gradle.launcher.daemon.server.exec.StartBuildOrRespondWithBusy$1.run(StartBuildOrRespondWithBusy.java:50)
	at org.gradle.launcher.daemon.server.DaemonStateCoordinator$1.run(DaemonStateCoordinator.java:297)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:46)
Caused by: java.lang.IllegalStateException: Expected configuration ':swaggerTemplate' to contain exactly one file, however, it contains 2 files.
	at org.gradle.api.internal.file.AbstractFileCollection.getSingleFile(AbstractFileCollection.java:62)
	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getSingleFile(Unknown Source)
	at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.getProperty(BeanDynamicObject.java:213)
	at org.gradle.internal.metaobject.BeanDynamicObject.getProperty(BeanDynamicObject.java:156)
	at org.gradle.internal.metaobject.CompositeDynamicObject.getProperty(CompositeDynamicObject.java:55)
	at org.gradle.internal.metaobject.AbstractDynamicObject.getProperty(AbstractDynamicObject.java:60)
	at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getProperty(Unknown Source)
	at org.hidetake.gradle.swagger.generator.ResolveSwaggerTemplate$_closure1.doCall(ResolveSwaggerTemplate.groovy:15)
	at org.hidetake.gradle.swagger.generator.ResolveSwaggerTemplate$_closure1.doCall(ResolveSwaggerTemplate.groovy)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:66)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:79)
	at org.gradle.api.internal.file.collections.BuildDependenciesOnlyFileCollectionResolveContext.add(BuildDependenciesOnlyFileCollectionResolveContext.java:79)
	at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitContents(DefaultConfigurableFileCollection.java:91)
	at org.gradle.api.internal.file.CompositeFileCollection.visitDependencies(CompositeFileCollection.java:166)
	at org.gradle.api.internal.file.collections.DefaultConfigurableFileCollection.visitDependencies(DefaultConfigurableFileCollection.java:97)
	at org.gradle.api.internal.file.CompositeFileCollection$1.visitDependencies(CompositeFileCollection.java:116)
	at org.gradle.api.internal.file.CompositeFileTree$FilteredFileTree.visitDependencies(CompositeFileTree.java:122)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext$TaskGraphImpl.getNodeValues(CachingTaskDependencyResolveContext.java:88)
	at org.gradle.internal.graph.CachingDirectedGraphWalker$GraphWithEmpyEdges.getNodeValues(CachingDirectedGraphWalker.java:202)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.doSearch(CachingDirectedGraphWalker.java:112)
	at org.gradle.internal.graph.CachingDirectedGraphWalker.findValues(CachingDirectedGraphWalker.java:64)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.doResolve(CachingTaskDependencyResolveContext.java:76)
	at org.gradle.api.internal.tasks.CachingTaskDependencyResolveContext.resolve(CachingTaskDependencyResolveContext.java:65)
	... 65 more


BUILD FAILED

Total time: 1.026 secs
Expected configuration ':swaggerTemplate' to contain exactly one file, however, it contains 2 files.
10:53:16 AM: Tasks execution finished 'clean build --stacktrace'.
```

--- a/src/main/groovy/org/hidetake/gradle/swagger/generator/ResolveSwaggerTemplate.groovy
+++ b/src/main/groovy/org/hidetake/gradle/swagger/generator/ResolveSwaggerTemplate.groovy
@@ -4,17 +4,13 @@ import org.gradle.api.tasks.Sync
 
 /**
  * A task to extract template files from a dependency.
- *
- * @author Hidetake Iwata
  */
 class ResolveSwaggerTemplate extends Sync {
 
-    def ResolveSwaggerTemplate() {
+    ResolveSwaggerTemplate() {
         from {
-            if (!project.configurations.swaggerTemplate.empty) {
-                project.zipTree(project.configurations.swaggerTemplate.singleFile)
-            } else {
-                []
+            project.configurations.swaggerTemplate.resolve().collect {
+                project.zipTree(it)
             }
         }
         into("${project.buildDir}/swagger-template")

--- a/src/test/groovy/org/hidetake/gradle/swagger/generator/Fixture.groovy
+++ b/src/test/groovy/org/hidetake/gradle/swagger/generator/Fixture.groovy
@@ -3,12 +3,33 @@ package org.hidetake.gradle.swagger.generator
 import org.gradle.api.Project
 import org.gradle.testfixtures.ProjectBuilder
 
+import java.util.zip.ZipEntry
+import java.util.zip.ZipOutputStream
+
 class Fixture {
     static Project projectWithPlugin(@DelegatesTo(Project) Closure closure = {}) {
         def project = ProjectBuilder.builder().build()
         project.apply(plugin: 'org.hidetake.swagger.generator')
         project.with(closure)
         project
+    }
+
+    /**
+     * Create a JAR file with given contents.
+     *
+     * @param destination JAR file
+     * @param contents map of filename and content
+     */
+    static void createJAR(File destination, Map<String, String> contents) {
+        destination.withOutputStream {
+            new ZipOutputStream(it).withStream { zip ->
+                contents.each { filename, content ->
+                    zip.putNextEntry(new ZipEntry(filename))
+                    zip << content
+                    zip.closeEntry()
+                }
+            }
+        }
     }
 
     static File file(YAML yaml) {

--- a/src/test/groovy/org/hidetake/gradle/swagger/generator/ResolveSwaggerTemplateSpec.groovy
+++ b/src/test/groovy/org/hidetake/gradle/swagger/generator/ResolveSwaggerTemplateSpec.groovy
@@ -1,0 +1,42 @@
+package org.hidetake.gradle.swagger.generator
+
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+import spock.lang.Specification
+
+class ResolveSwaggerTemplateSpec extends Specification {
+
+    @Rule
+    TemporaryFolder temporaryFolder = new TemporaryFolder()
+
+    def 'task should expand JARs into build/swagger-template'() {
+        given:
+        def repositoryFolder = temporaryFolder.newFolder()
+        Fixture.createJAR(new File(repositoryFolder, 'template-foo-1.0.0.jar'), [
+            'foo.txt': 'foo-hello-world',
+        ])
+        Fixture.createJAR(new File(repositoryFolder, 'template-bar-1.0.0.jar'), [
+            'bar.txt': 'bar-hello-world',
+        ])
+
+        def project = Fixture.projectWithPlugin {
+            repositories {
+                flatDir {
+                    dirs repositoryFolder
+                }
+            }
+            dependencies {
+                swaggerTemplate 'com.example:template-foo:1.0.0'
+                swaggerTemplate 'com.example:template-bar:1.0.0'
+            }
+        }
+
+        when:
+        project.tasks.resolveSwaggerTemplate.copy()
+
+        then:
+        project.file('build/swagger-template/foo.txt').text == 'foo-hello-world'
+        project.file('build/swagger-template/bar.txt').text == 'bar-hello-world'
+    }
+
+}

--- a/src/test/groovy/org/hidetake/gradle/swagger/generator/SwaggerGeneratorPluginSpec.groovy
+++ b/src/test/groovy/org/hidetake/gradle/swagger/generator/SwaggerGeneratorPluginSpec.groovy
@@ -25,6 +25,7 @@ class SwaggerGeneratorPluginSpec extends Specification {
         project.tasks.generateSwaggerUI
         project.tasks.generateReDoc
         project.tasks.validateSwagger
+        project.tasks.resolveSwaggerTemplate
 
         and: 'should have default properties'
         project.tasks.generateReDoc.scriptSrc =~ /\.js$/

